### PR TITLE
Dev/error handling

### PR DESCRIPTION
**Pull request by moneeroz**
- added error pages
- Implemented react-toastify library to handle error, warning, success or promise data that is displayed to the user

### CODE REVIEW

**Comment by moneeroz** [comment]
add a new user role page for users when they sign in for the first time.
the page will handle role update on first sign in.

**Review by mikhail-zapolskyi — APPROVED** [review]

**Review by mikhail-zapolskyi — APPROVED** [review]

File: app/dashboard/layout.tsx
[the remark this answers is not in the document]
Comment (moneeroz) [reply app/dashboard/layout.tsx:1]: will try it and see if it works

File: app/dashboard/page.tsx
[the remark this answers is not in the document]
Comment (moneeroz) [reply app/dashboard/page.tsx:1]: will delete

File: app/dashboard/layout.tsx
[the remark this answers is not in the document]
Comment (moneeroz) [reply app/dashboard/layout.tsx:1]: it worked. I only applied it to the dashboard layout because it is the one that contains auth routes

**Pull request merged**